# EDA — Precios de Musicales en Cartelera (Muestra Curada 2025–2026)

Dataset elaborado a partir de información publicada en las webs oficiales de musicales en cartelera. Recoge una muestra acotada, normalizada y verificada mediante conocimiento sectorial. Su objetivo es ofrecer una base de análisis sobre precios y estructura de la oferta actual.

**Input:** `precios_musicales_limpio_final.csv` (generado por `Eda_precios_musicales_limpio.ipynb`)  
**Output:** `data_processed/precios_musicales_limpio_definitivo.csv`


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


## 1. Carga y revisión inicial


In [2]:
# ============================
# 1. Carga del dataset
# ============================

df = pd.read_csv("../data_raw/precios_musicales_limpio_final.csv")

# ============================
# 2. Revisión inicial
# ============================

print("Shape:", df.shape)
print("\nColumnas:", df.columns.tolist())
print("\nInfo:")
print(df.info())
print("\nPrimeras filas:")


Shape: (8, 12)

Columnas: ['obra', 'fecha', 'día', 'precio_min', 'precio_max', 'ciudad', 'precio_estandar', 'productora', 'teatro', 'incluye_gastos', 'fuente ', 'diferencia_precios']

Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   obra                8 non-null      object 
 1   fecha               8 non-null      object 
 2   día                 8 non-null      object 
 3   precio_min          8 non-null      float64
 4   precio_max          8 non-null      float64
 5   ciudad              8 non-null      object 
 6   precio_estandar     8 non-null      float64
 7   productora          8 non-null      object 
 8   teatro              8 non-null      object 
 9   incluye_gastos      7 non-null      object 
 10  fuente              8 non-null      object 
 11  diferencia_precios  8 non-null      float64
dtypes: float64(4), obj

In [3]:
df.head(10)

,obra,fecha,día,precio_min,precio_max,ciudad,precio_estandar,productora,teatro,incluye_gastos,fuente,diferencia_precios
0,El rey león,2025-03-28,sábado,52.92,183.60,Madrid,131.76,Stage,Lope de vega,si,Entradas para El Rey León | Página Oficial,130.68
1,Cenicienta,2025-03-28,sábado,37.99,170.00,Madrid,105.00,Stage,Coliseum,si,Entradas de Cenicienta el Musical | Página Ofi...,132.01
2,Los pilares de la tierra,2025-01-31,sabádo,37.80,110.16,Madrid,73.44,Beón,EDP Gran Vía,si,"Los pilares de la Tierra, el musical | beon. e...",72.36
3,Los miserables,2025-03-28,sábado,49.90,120.00,Madrid,109.90,ATG/SOM,Nuevo Apolo,si,entradas.com,70.10
4,Wicked,2025-03-28,sábado,39.00,119.01,Madrid,103.00,ATG/SOM,Nuevo Alcalá,NaN,https://wickedelmusical.com/entradas/,80.01
5,We will rock you,2025-02-28,sábado,25.00,54.00,Madrid,54.00,GT CXABANK PP,La estación PP,si,"Inicio - WE WILL ROCK YOU, EL MUSICAL",29.00
6,Mamma mia!,2025-03-28,sábado,55.00,75.00,Gira,69.00,ATG/SOM,La Laboral (GiJón),si,MAMMA MÍA! El Musical | Gira Nacional | Página...,20.00
7,"The book of mormon, el musical",2025-01-31,sábado,35.39,86.20,Madrid,77.13,ATG/SOM,Calderón,si,Entradas – THE BOOK OF MORMON El Musical,50.81


In [4]:
# Añadir filas de precios registrados posteriormente
nuevas_filas = pd.DataFrame([
    {
        "obra": "Cabaret",
        "fecha": "2026-03-28",
        "día": "sábado",
        "precio_min": 39.2,
        "precio_max": 155.6,
        "ciudad": "Madrid",
        "precio_estandar": 68.0,
        "productora": "Let's Go",
        "teatro": "Albéniz",
        "incluye_gastos": "si",
        "fuente": "entradas.com",
        "diferencia_precios": 116.4
    },
    {
        "obra": "101 Dálmatas",
        "fecha": "2026-03-27",
        "día": "viernes",
        "precio_min": 28.5,
        "precio_max": 58.4,
        "ciudad": "Madrid",
        "precio_estandar": 58.4,
        "productora": "GT CXABANK PP Príncipe Pío",
        "teatro": "La estación PP",
        "incluye_gastos": "si",
        "fuente": "entradas.com",
        "diferencia_precios": 29.9
    }
])

df = pd.concat([df, nuevas_filas], ignore_index=True)


In [5]:
df.tail()

,obra,fecha,día,precio_min,precio_max,ciudad,precio_estandar,productora,teatro,incluye_gastos,fuente,diferencia_precios,fuente
5,We will rock you,2025-02-28,sábado,25.00,54.0,Madrid,54.00,GT CXABANK PP,La estación PP,si,"Inicio - WE WILL ROCK YOU, EL MUSICAL",29.00,NaN
6,Mamma mia!,2025-03-28,sábado,55.00,75.0,Gira,69.00,ATG/SOM,La Laboral (GiJón),si,MAMMA MÍA! El Musical | Gira Nacional | Página...,20.00,NaN
7,"The book of mormon, el musical",2025-01-31,sábado,35.39,86.2,Madrid,77.13,ATG/SOM,Calderón,si,Entradas – THE BOOK OF MORMON El Musical,50.81,NaN
8,Cabaret,2026-03-28,sábado,39.20,155.6,Madrid,68.00,Let's Go,Albéniz,si,NaN,116.40,entradas.com
9,101 Dálmatas,2026-03-27,viernes,28.50,58.4,Madrid,58.40,GT CXABANK PP Príncipe Pío,La estación PP,si,NaN,29.90,entradas.com


In [6]:
df.shape

(10, 13)

In [7]:
df["fecha"] = pd.to_datetime(df["fecha"])

In [8]:
df.dtypes

obra                          object
fecha                 datetime64[ns]
día                           object
precio_min                   float64
precio_max                   float64
ciudad                        object
precio_estandar              float64
productora                    object
teatro                        object
incluye_gastos                object
fuente                        object
diferencia_precios           float64
fuente                        object
dtype: object

In [9]:
df.columns = df.columns.str.strip()
df = df.loc[:, ~df.columns.duplicated()]

In [10]:
df = df.drop(columns=["incluye_gastos","fuente"])

In [11]:
df.columns

Index(['obra', 'fecha', 'día', 'precio_min', 'precio_max', 'ciudad',
       'precio_estandar', 'productora', 'teatro', 'diferencia_precios'],
      dtype='object')

## 2. Calidad, estadísticos y visualizaciones


In [12]:
# A?adir id_obra para enlace con dim_obra en Power BI
dim_obra_ref = pd.read_csv("../data_processed/dim_obra.csv")[["id_obra", "obra"]]
dim_obra_ref["obra_key"] = dim_obra_ref["obra"].str.strip().str.lower()
df["obra_key"] = df["obra"].str.strip().str.lower()
# Normalizar nombre del Book of Mormon (sufijo distinto entre datasets)
df["obra_key"] = df["obra_key"].str.replace(", el musical", "", regex=False)
df = df.merge(dim_obra_ref[["id_obra", "obra_key"]], on="obra_key", how="left").drop(columns="obra_key")
# Asignacion directa para 101 Dalmatians (variante de nombre sin sufijo)
df.loc[df["id_obra"].isna() & df["obra"].str.contains("101", na=False), "id_obra"] = 1

# Exportacion final
df.to_csv(
    "../data_processed/precios_musicales_limpio_definitivo.csv",
    index=False, sep=",", decimal="."
)
print("EDA completado y dataset exportado.")
print(df[["obra","id_obra"]].to_string())


EDA completado y dataset exportado.
                             obra  id_obra
0                     El rey león     20.0
1                      Cenicienta      9.0
2        Los pilares de la tierra     43.0
3                  Los miserables     42.0
4                          Wicked     67.0
5                We will rock you     66.0
6                      Mamma mia!     47.0
7  The book of mormon, el musical     60.0
8                         Cabaret      7.0
9                    101 Dálmatas      1.0


## Conclusiones — Precios del teatro musical

- Dataset final de 10–11 registros, sin duplicados y con variables tipadas correctamente.
- El precio estándar se concentra entre 70 € y 120 €, con dispersión moderada.
- Los mínimos son relativamente estables (~30–55 €); los máximos son más variables (~60–180 €), indicando segmentación por categoría de asiento.
- *El Rey León*, *Los Miserables* y *Cenicienta/Wicked* presentan los precios estándar más elevados; *Mamma Mia!* y *We Will Rock You* se sitúan en el tramo inferior.
- Las producciones premium muestran mayores brechas min/max, reflejando estrategias comerciales más agresivas.
- Los teatros asociados a grandes producciones (Lope de Vega, Coliseum, Nuevo Apolo) concentran los precios más altos.